scripts/pack.js

```diff
@@ -596,24 +596,36 @@ function packScripts() {
   execSync(`chmod +x ${binDir}/\*.sh`);
 }
 
-function packSamples() {
-  if (!options['sample-path']) return;
+function packApps() {
+  if (!options['app-path']) return;
   chdir(originCwd);
-  var samplePath = options['sample-path'];
-  if (!fs.existsSync(samplePath)) {
-    console.log(`\x1b[31mError: ${samplePath} does not exist\x1b[0m`);
+  var appPath = options['app-path'];
+  if (!fs.existsSync(appPath)) {
+    console.log(`\x1b[31mError: ${appPath} does not exist\x1b[0m`);
     return;
   }
-  execSync(`rm -rf ${distDir}/extras`);
-  execSync(`mkdir -p ${distDir}/extras`);
-  execSync(`cp -a ${samplePath} ${distDir}/extras/basic_example`);
+  execSync(`rm -rf ${distDir}/apps`);
+  execSync(`mkdir -p ${distDir}/apps`);
+  console.log('\x1b[32mApps folder created in :', distDir, '\x1b[0m');
+  execSync(`cp -a ${appPath} ${distDir}/apps/current_app`);
 
-  const certScript = `${distDir}/extras/basic_example/initcert.js`;
+  // Look in the app's package.json to see what file to use for main.js
+  var jsonTXT = execSync(`cat ${distDir}/apps/current_app/package.json`);
+  var appJSON = JSON.parse(jsonTXT)["main"];
+
+  if (!appJSON === undefined) {
+    console.log("\x1b[31mError: No main js file for the app\x1b[0m");
+    return;
+  } else {
+    // Make a soft link to the main JS file node.js should call
+    execSync(`ln -sf ${distDir}/apps/current_app/${appJSON} ${distDir}/apps/current_app/main.js`);
```

Comment:
What if ${appJSON} is "main.js", will it affect packing?

Reply:
Good point, yes the link would fail and it would throw an error. I'll add a check and skip this step if the file is called main.js.